yt_dlp/extractor/nintendo.py

```diff
@@ -1,57 +1,101 @@
-import re
+import json
+from urllib.parse import quote
 
 from .common import InfoExtractor
-from .ooyala import OoyalaIE
+from ..utils import (
+    ExtractorError,
+    make_archive_id,
+    remove_start,
+    unified_timestamp,
+    urljoin,
+)
+from ..utils.traversal import traverse_obj
 
 
 class NintendoIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?nintendo\.com/(?:games/detail|nintendo-direct)/(?P<id>[^/?#&]+)'
+    _VALID_URL = r'https?://(?:www\.)?nintendo\.com/(?:(?P<locale>\w{2}(?:-\w{2})?)/)?(nintendo-direct)/(?P<id>[^/?#&]+)'
     _TESTS = [{
-        'url': 'https://www.nintendo.com/games/detail/duck-hunt-wii-u/',
-        'info_dict': {
-            'id': 'MzMmticjp0VPzO3CCj4rmFOuohEuEWoW',
-            'ext': 'flv',
-            'title': 'Duck Hunt Wii U VC NES - Trailer',
-            'duration': 60.326,
-        },
-        'params': {
-            'skip_download': True,
-        },
-        'add_ie': ['Ooyala'],
-    }, {
-        'url': 'http://www.nintendo.com/games/detail/tokyo-mirage-sessions-fe-wii-u',
+        'url': 'https://www.nintendo.com/nintendo-direct/09-04-2019/',
         'info_dict': {
-            'id': 'tokyo-mirage-sessions-fe-wii-u',
-            'title': 'Tokyo Mirage Sessions ♯FE',
+            'ext': 'mp4',
+            'id': '2oPmiviVePUA1IqAZzjuVh',
+            'display_id': '09-04-2019',
+            'title': 'Nintendo Direct 9.4.2019',
+            'timestamp': 1567580400,
+            'description': 'md5:8aac2780361d8cb772b6d1de66d7d6f4',
+            'upload_date': '20190904',
+            'age_limit': 17,
+            '_old_archive_ids': ['nintendo J2bXdmaTE6fe3dWJTPcc7m23FNbc_A1V'],
         },
-        'playlist_count': 4,
     }, {
-        'url': 'https://www.nintendo.com/nintendo-direct/09-04-2019/',
+        'url': 'https://www.nintendo.com/en-ca/nintendo-direct/08-31-2023/',
         'info_dict': {
-            'id': 'J2bXdmaTE6fe3dWJTPcc7m23FNbc_A1V',
             'ext': 'mp4',
-            'title': 'Switch_ROS_ND0904-H264.mov',
-            'duration': 2324.758,
-        },
-        'params': {
-            'skip_download': True,
+            'id': '2TB2w2rJhNYF84qQ9E57hU',
+            'display_id': '08-31-2023',
+            'title': 'Super Mario Bros. Wonder Direct 8.31.2023',
+            'timestamp': 1693465200,
+            'description': 'md5:3067c5b824bcfdae9090a7f38ab2d200',
+            'tags': ['Mild Fantasy Violence', 'In-Game Purchases'],
+            'upload_date': '20230831',
+            'age_limit': 6,
         },
-        'add_ie': ['Ooyala'],
+    }, {
+        'url': 'https://www.nintendo.com/en/nintendo-direct/09-04-2019/',
+        'only_matching': True,
     }]
 
+    def _create_asset_url(self, path):
+        return urljoin('https://assets.nintendo.com/', quote(path))
+
     def _real_extract(self, url):
-        page_id = self._match_id(url)
+        locale, video_id = self._match_valid_url(url).group('locale', 'id')
```

Comment:
Maybe for clarity this could be `display_id` here already? I originally wanted to make this the `id` but that seemed to rudimentary since we have access to the internal media id